bigtable/client/metadata_update_policy_test.cc

```diff
@@ -13,13 +13,278 @@
 // limitations under the License.
 
 #include "bigtable/client/metadata_update_policy.h"
+#include "bigtable/client/admin_client.h"
+#include "bigtable/client/internal/make_unique.h"
+#include "bigtable/client/table.h"
+#include "bigtable/client/table_admin.h"
+#include "google/bigtable/v2/bigtable.grpc.pb.h"
+#include <google/bigtable/admin/v2/bigtable_table_admin.grpc.pb.h>
 #include <gtest/gtest.h>
+#include <thread>
 
 std::string const kInstanceName = "projects/foo-project/instances/bar-instance";
 std::string const kTableId = "baz-table";
 std::string const kTableName =
     "projects/foo-project/instances/bar-instance/tables/baz-table";
 
+namespace btproto = google::bigtable::v2;
+namespace adminproto = google::bigtable::admin::v2;
+
+/**
+ * Implement the portions of the `google.bigtable.v2.Bigtable` interface
+ * necessary for the embedded server tests.
+ *
+ * This is not a Mock (use `google::bigtable::v2::MockBigtableStub` for that,
+ * nor is this a Fake implementation (use the Cloud Bigtable Emulator for that),
+ * this is an implementation of the interface that returns hardcoded values.
+ * It is suitable for the embedded server tests, but for nothing else.
+ */
+class BigtableImpl final : public btproto::Bigtable::Service {
+ public:
+  BigtableImpl() {}
+  grpc::Status ReadRows(
+      grpc::ServerContext* context,
+      google::bigtable::v2::ReadRowsRequest const* request,
+      grpc::ServerWriter<google::bigtable::v2::ReadRowsResponse>* writer)
+      override {
+    for (std::multimap<grpc::string_ref, grpc::string_ref>::const_iterator it =
```

Comment:
Also, why `grpc::string_ref` is dangerous.  Please just copy into `std::string`.  That is safe and we do not need to avoid all memory copying in the test.

Reply:
converted string_Ref to string